js/components/Feed.jsx

```diff
@@ -55,10 +55,11 @@ var Feed = React.createClass({
         this.setState({ transactions : transactions.slice(0,15), pagination: 0 });
 
         // If we have less than the pagination size + 1, then there are no more pages to fetch
+        // If there are no more pages to fetch, then 0 is the final pagination index
         if (transactions.length < PAGINATION_SIZE + 1) {
-          this.setState({ endOfResults: true });
+          this.setState({ finalPaginationIndex: 0 });
         } else {
-          this.setState({ endOfResults: false });
+          this.setState({ finalPaginationIndex: Number.POSITIVE_INFINITY });
```

Comment:
Instead of a boolean, capture the actual final pagination index if we ever see it.